lib/workshops.dart

```diff
@@ -318,9 +381,38 @@ class WorkshopUi extends EditorUi {
         });
       });
 
+    revertButton = MDCButton(querySelector('#revert-button') as ButtonElement,
+        isIcon: true)
+      ..setAttr('hidden', 'true')
+      ..onClick.listen((_) {
+        // Go back to the original snippet code.
+        if (fullDartSource != _workshopState.currentStep.snippet) {
+          editor.document.updateValue(_workshopState.currentStep.snippet);
+        }
+      });
+
+    redoButton =
+        MDCButton(querySelector('#redo-button') as ButtonElement, isIcon: true)
+          ..setAttr('hidden', 'true')
+          ..onClick.listen((_) {
+            // Go back to the users saved code.
+            final String? usersWork = _workshopStepStorage
+                .getStoredWorkForStep(_workshopState.currentStepIndex);
+            if (usersWork != null) {
+              editor.document.updateValue(usersWork);
+            }
+          });
+    showSolutionIconButton = MDCButton(
```

Comment:
I'm not sure we need this button, it might be more confusing to the user if there's two buttons that do the same thing.